pkg/provider/userinterface.go

```diff
@@ -0,0 +1,12 @@
+package provider
```

Comment:
Bad (very) file & package name. How do you expect others to understand package purpose?